### PR TITLE
Do not create doc test folder out of /test/tmp

### PR DESCRIPTION
**Pull request by eksperimental**
* docs/ renamed to doc/ to follow elixir convention

### CODE REVIEW

File: test/ex_doc/formatter/html_test.exs [hunk 1]
```diff
@@ -5,10 +5,10 @@ defmodule ExDoc.Formatter.HTMLTest do
 
   setup_all do
     File.mkdir(output_dir)
-    File.copy("test/fixtures/README.md", "test/README.md")
+    File.copy("test/fixtures/README.md", "test/tmp/README.md")
 
     on_exit fn ->
-      File.rm("test/README.md")
+      File.rm("test/tmp/README.md")
       System.cmd "rm", ["-rf", "#{output_dir}"]
```
Comment (eksperimental) [inline test/ex_doc/formatter/html_test.exs]: I'm  also in favour of getting rid of these two lines, so we can actually see the output when debugging exdoc.
Anyways they will be removed on `mix clean`

Comment (josevalim) [inline test/ex_doc/formatter/html_test.exs]: We need to clean up at some point though, at most before the test starts. We can also use `File.rm_rf` now.

Comment (eksperimental) [inline test/ex_doc/formatter/html_test.exs]: true, yeah, when the test starts sounds perfect to me

**Comment by eksperimental** [comment]
let me know what you think @josevalim 
I have added a few tests as well. let me know if we should add a test for any other situation

**Comment by eksperimental** [comment]
Any idea why the build is failing?
> ** (Mix) Expected :only in dependency to be an atom, got: [:test]
> The command "mix do local.hex --force, deps.get && mix test --include pandoc" exited with 1.
it's also happening with the other PR

**Comment by josevalim** [comment]
It is a bug in Elixir I have just fixed.

**Comment by eksperimental** [comment]
:+1: great
thank you

**Pull request merged**